payload/builder.go

```diff
@@ -324,6 +326,58 @@ func (p *Payload) SoundVolume(volume float32) *Payload {
 	return p
 }
 
+// InterruptionLevelPassive sets the aps interruption-level to "passive" on the payload.
+// This is to indicate the importance and delivery timing of a notification.
+// See: https://developer.apple.com/documentation/usernotifications/unnotificationinterruptionlevel/
+//
+// {"aps":{"interruption-level":passive}}
+func (p *Payload) InterruptionLevelPassive() *Payload {
+	p.aps().InterruptionLevel = "passive"
+	return p
+}
+
+// InterruptionLevelActive sets the aps interruption-level to "active" on the payload.
+// This is to indicate the importance and delivery timing of a notification.
+// See: https://developer.apple.com/documentation/usernotifications/unnotificationinterruptionlevel/
+//
+// {"aps":{"interruption-level":active}}
+func (p *Payload) InterruptionLevelActive() *Payload {
+	p.aps().InterruptionLevel = "active"
+	return p
+}
+
+// InterruptionLevelTimeSensitive sets the aps interruption-level to "time-sensitive" on the payload.
+// This is to indicate the importance and delivery timing of a notification.
+// See: https://developer.apple.com/documentation/usernotifications/unnotificationinterruptionlevel/
+//
+// {"aps":{"interruption-level":time-sensitive}}
+func (p *Payload) InterruptionLevelTimeSensitive() *Payload {
+	p.aps().InterruptionLevel = "time-sensitive"
+	return p
+}
+
+// InterruptionLevelCritical sets the aps interruption-level to "critical" on the payload.
+// This is to indicate the importance and delivery timing of a notification.
+// This interruption level requires an approved entitlement from Apple.
+// See: https://developer.apple.com/documentation/usernotifications/unnotificationinterruptionlevel/
+//
+// {"aps":{"interruption-level":critical}}
+func (p *Payload) InterruptionLevelCritical() *Payload {
+	p.aps().InterruptionLevel = "critical"
+	return p
+}
+
```

Comment:
Is there a reason to use these four different functions for setting this value rather than a single setter (i.e. `InterruptionLevel`) and define a custom string type ala `EPushType` with the values as defined as constants?

Reply:
I used the four functions as I was following the way that individual functions were provided for badges eg. `ZeroBadge` etc.

I am happy to swap to a single setter (which was my original thought tbh).

Welcome your thoughts @chrishaines?

Reply:
I'm not sure but it seems to be that this is partially due to typing (it allows `Badge` to specifically take an `int` as an argument with `UnsetBadge` being used to set it to `nil` to remove it from the payload. I'm not sure why there is a separate `ZeroBadge` function rather than just using `Badge(0)` 🤷‍♂️ ). 

I think that `EPushType` seems to be the closest precedent to handling an enumerated value in the codebase.

Reply:
I am inclined to swap it out then as you suggest (I have working code for that implementation ready to go).

Before I do though, can I ask, what is the reason for the `E` prefix in `EPushType`? Is there some specific naming convention? (I don't do a huge lot in Go).

Reply:
I'm not sure where that came from either to be honest. Perhaps it's for `Enum` (complete guess)?

Reply:
I think that seems a sensible guess!
Just wondering if I should employ the same `E` prefix or not!

Reply:
I think that sounds reasonable.

Reply:
I have swapped out the InterruptionLevel into a single function. Thanks for your input @chrishaines.